### PR TITLE
Refactor/Add methods and make it packagecloud ready.

### PR DESCRIPTION
**Pull request by MansurTsutiev**
## Why do we need this change?
- We want `data-source.js` to live on packagecloud.
- We also want to refactor/add some methods to make it more dev-friendly.
(Not %100 sure if the refactor makes it any more dev-friendly, open to suggestions)

## Implementation Details
See README.md

#### Dependencies (if any)
none

:house: [ch43507](https://app.clubhouse.io/movableink/story/43507)

### CODE REVIEW

**Comment by aqmnguyen** [comment]
Should we take this opportunity to rename `getAllRows` to something else? It's somewhat potentially mis-leading as it doesn't get you back _all_ the rows. 

For any new users of this library they could question why they would even use `getSingleTarget` if `getAllRows` is there. 

Was also thinking this would be a good chance to make this version `1.0.0` since it goes into PackageCloud and since it's a major release we _should_ be okay to rename the methods available in the lib

File: src/index.ts [hunk 1]
```diff
@@ -53,19 +62,59 @@ export default class DataSource {
    *
    * @param opts
    */
-  async getMultipleTargets(opts: any = {}) {
+  async getMultipleTargets(sets: any = [], opts: any = {}) {
```
Comment (aqmnguyen) [inline src/index.ts:65]: Doesn't this param addition mean that the current way SD's are using this method has to somewhat change.
As the release we did in `0.0.3` we wanted to make it consistent to how we did `CD.getCORS`. 

Which was something like this: 

```js
const options = {
  method: 'POST',
  body: JSON.stringify([{}])
}
getMultipleTargets(options)
```

Would it make more sense to make `sets` the second argument and making it optional?

Comment (MansurTsutiev) [reply src/index.ts:65]: Yeah that makes sense. I'll revert it to how it was before (just taking in a single arg `options`).
I'm not sure that having `sets` as the second optional arg adds any value, do we even need it then?

Comment (MansurTsutiev) [reply src/index.ts:65]: Should I also update `getSingleTarget` to also just take a single arg `options`?

**Comment by MansurTsutiev** [comment]
What about renaming `getAllRows` to `getMultipleRows` ?

File: src/index.ts (hunk 1 above)
Comment (aqmnguyen) [reply src/index.ts:65]: Yea, I wasn't sure if you needed it for something else. 
I do agree keeping it as a single arg `options` to make it similar to how we doing `getMultipleTargets` would be better for consistency

**Comment by aqmnguyen** [comment]
> What about renaming `getAllRows` to `getMultipleRows` ?

Without being super pedantic on this, what if someone only wants 1 row. 
Just curious on this, we even need this method? Just thinking as `getSingleTarget` does the same thing but just using the new `POST` request vs the old `GET`? 

Is there any other benefit to this `GET` that i'm not thinking about?

**Comment by MansurTsutiev** [comment]
> Is there any other benefit to this `GET` that i'm not thinking about?

I don't think there is, the only difference is that you can pass extra query params to `getAllRows` but we can update new methods to take in params too. If there are no objections I'll remove `getAllRows`.

**Review by aqmnguyen — COMMENTED** [review]
Code wise, everything looks good. 
The only comment I have is that the read me needs updating as it still has `getAllRows` there and  the geo-location example says to use `getAllRows` but further down there is the new location method.

**Comment by MansurTsutiev** [comment]
Hey @aqmnguyen, I paired with Dennis today and we managed to fix the error (👏 @shyshy ). Test it out again, let me know if everything looks good. You can test it by `yarn link`ing it or by including 
`"@movable-internal/data-source.js": "1.0.0"` in your `package.json` 
(just to also make sure it works for you by pulling it down from packagecloud).

**Review by aqmnguyen — APPROVED** [review]
Tried it and all looks good 👍

File: karma.conf.js [hunk 2]
```diff
@@ -1,63 +1,52 @@
...
-// Generated on Tue Nov 21 2017 17:02:58 GMT-0500 (EST)
+const isDocker = require('is-docker');
+const watchMode = process.env.KARMA_WATCH === 'true' || true;
+const babel = require('rollup-plugin-babel');
+const resolve = require('rollup-plugin-node-resolve');
+const commonjs = require('@rollup/plugin-commonjs');
+const babelConf = require('./babel-config');
 
-module.exports = function(config) {
+module.exports = function kamraConfig(config) {
```
Comment (aqmnguyen) [inline karma.conf.js:8]: Actually just noticed this typo (?)

**Comment by MansurTsutiev** [comment]
Thanks for updating the readme 👍

**Review by aqmnguyen — APPROVED** [review]

**Review by shyshy — APPROVED** [review]

**Pull request merged**